### ISSUE TITLE
[bug] Conan error on 1:st connection to server after reboot

### ISSUE THREAD

**Issue by foobar222** [conan-io/conan#9730]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: N/A
  * Conan version: 1.39.0
  * Python version: 3.8.3

### Steps to reproduce (Include if Applicable)

We use Conan to install dependencies for unattended nightly builds. This usually works perfectly, with one exception: the first build after rebooting the build machine always fails on `conan install`

 1. reboot the computer
 2. set CONAN_NON_INTERACTIVE=1
 3. run conan install 

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
conan install -u -if build /path/to/conanfile.txt
--- SNIP ---
DEBUG :rest_client_common.py[160]: REST: ping: https://[redacted]/v1/ping [2021-10-04 20:33:33,001]
DEBUG :rest_client.py [58]: REST: Cached capabilities for the remote: ['complex_search', 'revisions', 'revisions'] [2021-10-04 20:33:33,115]
DEBUG :rest_client_common.py[188]: REST: get: https://[redacted]/v1/conans/[redacted]/stable/digest [2021-10-04 20:33:33,117]
DEBUG :rest_client_common.py[30]: REST ERROR: <class 'conans.errors.AuthenticationException'> [2021-10-04 20:33:33,119]
INFO  :auth_manager.py[63]: Token expired or not valid, cleaning the saved token and retrying [2021-10-04 20:33:33,123]
ERROR: Conan interactive mode disabled. [Remote: XXXX]
```

There are no problems with subsequent invocations of `conan install`, only the first attempt after a reboot fail, consistently. There are no indications of errors on the conan server.

**Comment by memsharded** [comment]
Hi @foobar222 

Quick question, which server are you using (and version, just in case)?

**Comment by foobar222** [comment]
conan_server, version 1.37.0, on Ubuntu 18.04.3 LTS.  It's configured to allow unauthenticated downloads:
```
[read_permissions]
*/*@*/*: *
```

**Comment by memsharded** [comment]
I am assuming that the ``conan_server`` is running in a different machine than the client one that is being rebooted?

**Comment by foobar222** [comment]
Yes.

**Comment by lasote** [comment]
Hi, the server tokens are expiring by default in 30 minutes (1800 seconds). When the token expires it request again authentication to the user causing the error (because of the non-interactive).
Might be possible that the token is expiring?
You can adjust the `authorize_timeout` at the server config: https://docs.conan.io/en/latest/uploading_packages/running_your_server.html

**Comment by foobar222** [comment]
Well, the client output says token has expired, so I do believe that is happening. I'm not sure that is the issue, though. The client computer is rebooted once a week, on schedule. The command `conan install` is invoked several times every night, connecting to the conan server. At least once every 24 hours, the token expiration timeout is exceeded, usually without any problems at all, with one exception: the first invocation of `conan install` after a reboot.

I would expect the conan client to discard any expired token and retry, but if I were to interpret the output, apparently the client fails with that because interactive mode is disabled. 

You have a great product, and this is not a serious issue, just a little annoying. Reported because I had the opportunity to observe a pattern. 

/J,

**Comment by lasote** [comment]
This is very unexpected. I made a simple test configuring `read_permissions` to `*/*@*/*: *` and I checked no authentication is required at any time installing packages. So:

- No token should be ever stored in the local database.
- BUT It makes no sense that the client receives an `AuthenticationException` if no token is sent to the server and the server is "open" to read.

So, This is kind of Sherlock Holmes guessing but I can only think that during the reboot something is messing with the local database. Is the conan cache always stored in the same location in a hard disk right? You are not running any scheduled Conan command or something similar, right? 

It is a bit inconvenient but, could you, next time you reboot, check the `~/.conan/.conan.db` file? that is a sqlite database, the `users_remotes` table should be empty.

**Comment by foobar222** [comment]
Updated and simplified steps to reproduce (there were other things going on that I wasn't aware of first, thanks for the hint on `users_remote` table):

1. Server configured to allow unauthenticated download, and require authentication for all uploads:
```
[read_permissions]
*/*@*/*: *
[write_permissions]
*/*@*/*: ?
```
2. Build and upload a package, using authentication with environment variables `CONAN_LOGIN_USERNAME` and `CONAN_PASSWORD`. This step succeeds, and the `token` column in `users_remote` table is set to a non-empty value.
3. Wait and allow the access token to expire.
4. set `CONAN_NON_INTERACTIVE=1`, clear `CONAN_LOGIN_USERNAME` and `CONAN_PASSWORD` environment variables.
5. try to install a package from the server, without authentication. The 1:st attempt fails (but seems to clear the  `token` column in `users_remote` table). 2:nd attempt of issuing the very same command succeeds.

**Comment by memsharded** [comment]
Thanks very much for your details to reproduce, it really did help.
I managed to write a failing test for it:

```python
def test_token_expired():
    server_folder = temp_folder()
    server_conf = textwrap.dedent("""
       [server]
       jwt_expire_minutes: 1
       authorize_timeout: 0
       disk_authorize_timeout: 0
       disk_storage_path: ./data
       updown_secret: 12345
       jwt_secret: mysecret
       port: 12345
       [read_permissions]
       */*@*/*: *
       [write_permissions]
       */*@*/*: admin
       """)
    save(os.path.join(server_folder, ".conan_server", "server.conf"),
         server_conf)
    server = TestServer(base_path=server_folder, users={"admin": "password"})

    c = TestClient(servers={"default": server}, users={"default": [("admin", "password")]})
    c.save({"conanfile.py": GenConanfile()})
    c.run("create . pkg/0.1@user/stable")
    c.run("upload * -r=default --all -c")
    print(c.out)
    print(c.cache.localdb.get_login(server.fake_url))

    import time
    time.sleep(61)
    c.users = {}
    c.run("config set general.non_interactive=1")
    c.run("remove * -f")
    c.run("install pkg/0.1@user/stable", assert_error=True)
    print(c.out)
    print(c.cache.localdb.get_login(server.fake_url))
    assert "Conan interactive mode disabled" in c.out

    # next one will run
    c.run("install pkg/0.1@user/stable")
```
A bit rough with the time.sleep (we shouldn't put that in our test suite), but at least it fails. Lets try to understand now what is happening.

**Comment by memsharded** [comment]
Doing an exploratory PR in https://github.com/conan-io/conan/pull/9781, to see what could break

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9781 seems to be ok, trying to move this forward for next Conan 1.42

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9781 merged, this will be released in 1.42
